### PR TITLE
[JENKINS-44663] Authorization header should be case insensitive

### PR DESCRIPTION
**Pull request by daurnimator**

### CODE REVIEW

File: core/src/main/java/jenkins/security/BasicHeaderProcessor.java [hunk 1]
```diff
@@ -60,7 +60,7 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         HttpServletResponse rsp = (HttpServletResponse) response;
         String authorization = req.getHeader("Authorization");
 
-        if (authorization!=null && authorization.startsWith("Basic ")) {
+        if (authorization!=null && authorization.regionMatches(true, 0, "Basic ", 0, 6)) {
```
Comment (daniel-beck) [inline core/src/main/java/jenkins/security/BasicHeaderProcessor.java]: `StringUtils.startsWithIgnoreCase` for readability?

File: test/src/test/java/jenkins/security/BasicHeaderProcessorTest.java [hunk 2]
```diff
@@ -80,16 +85,20 @@ private void makeRequestAndFail(String userAndPass) throws IOException, SAXExcep
         }
     }
 
-    private void makeRequestWithAuthAndVerify(String userAndPass, String username) throws IOException, SAXException {
+    private void makeRequestAndVerify(String headerName, String headerValue, String username) {
```
Comment (daniel-beck) [inline test/src/test/java/jenkins/security/BasicHeaderProcessorTest.java]: Still needs to declare `IOException, MalformedURLException` (PR build failure)

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Yes, needs a fix

File: core/src/main/java/jenkins/security/BasicHeaderProcessor.java (hunk 1 above)
Comment (daurnimator) [inline core/src/main/java/jenkins/security/BasicHeaderProcessor.java]: Is that a standard function? I didn't find much when I searched for it.

Comment (daniel-beck) [inline core/src/main/java/jenkins/security/BasicHeaderProcessor.java]: @daurnimator It's from commons-lang and included with Jenkins.

**Comment by daniel-beck** [comment]
> `BasicHeaderProcessorTest.java:[99,46] error: incompatible types: String cannot be converted to boolean`

@daurnimator 

Could you please at least compile the tests locally? Otherwise this becomes a waste of time for us.

**Comment by daurnimator** [comment]
I didn't manage to get tests to run locally. I'll let someone else fix the issue.

**Comment by oleg-nenashev** [comment]
@daurnimator would be great to know what was the issue with running tests

**Pull request closed without merging**